src/controllers/passwordRecoveryController.js

```diff
@@ -0,0 +1,88 @@
+const { BadRequestError, UnauthenticatedError } = require('../errors');
+const User = require('../models/User');
+const sendEmail = require('../utils/sendEmail');
+const jwt = require('jsonwebtoken');
+const { StatusCodes } = require('http-status-codes');
+
+const requestPasswordReset = async (req, res, next) => {
+  try {
+    const { email } = req.body;
+    if (!email) {
+      throw new BadRequestError('Please provide an email.');
+    }
+
+    const user = await User.findOne({ email });
+    if (!user) {
+      throw new BadRequestError('Invalid Credentials.');
+    }
+
+    const resetToken = jwt.sign({ userId: user._id }, process.env.JWT_SECRET, {
+      expiresIn: '1h',
+    });
+
+    console.log(resetToken);
+
+    user.resetToken = resetToken;
+    user.resetTokenExpires = Date.now() + 60 * 60 * 1000;
+    await user.save();
+
+    const resetLink = `${process.env.FRONTEND_URL}/password/edit?token=${resetToken}`;
+    if (!process.env.FRONTEND_URL) {
+      throw new BadRequestError('Frontend URL is not available in .env');
+    }
+
+    const resetMessage = `
+            <h3>Password Reset Request</h3>
+            <p>You are receiving this email because you (or someone else) have requested the reset of a password.</p>
+            <p>If you did not request this, please ignore this email.</p>
+            <p>Please click the link below to reset your password.</p>
+            <p>This link is valid for 1 hour:</p>
+            <a href="${resetLink}">Reset Password</a>
+          `;
+
+    await sendEmail(user.email, 'Password Reset Request', resetMessage);
```

Comment:
We should save the resetToken in the DB after the user has succesfully gotten an email, or else we might update it without them getting an email. 

await sendEmail(user.email, 'Password Reset Request', resetMessage);
user.resetToken = resetToken;
user.resetTokenExpires = Date.now() + 60 * 60 * 1000;
await user.save();